darts/utils/data/tabularization.py

```diff
@@ -197,24 +281,27 @@ def _add_static_covariates(model, series, features):
             # for when series in prediction do not have static covariates but some of the training series did
             pad_zeros = np.zeros((1, model.n_features_in_ - features.shape[1]))
             return np.concatenate(
-                [features, np.tile(pad_zeros, reps=(reps, 1))], axis=1
+                [features, np.tile(pad_zeros, reps=(features.shape[0], 1))], axis=1
             )
         else:
             return features
 
     else:
         # at least one series in the sequence has static covariates
         static_covs = []
-
+        col_names = list(scovs_map["names"])
         # build static covariates array
-        for i in range(len(series)):
-            pad_zeros = np.zeros((1, max_width - map["covs_width"][i]))
-            scovs = (
-                np.concatenate((map["values"][i], pad_zeros), axis=1)
-                if map["covs_width"][i] > 0
-                else pad_zeros
-            )
-            static_covs.append(np.tile(scovs, reps=(reps, 1)))
+        for i in range(len(target_series)):
+            df = scovs_map["df"][i]
+
+            if not df.empty:
+                df = df.reindex(col_names, axis=1, fill_value=0.0)
```

Comment:
Please make sure this is not affecting the original DF inside the user-provided TimeSeries (otherwise copy it).